cpp/roaring64map.hh

```diff
@@ -1092,6 +1093,29 @@ public:
         return result;
     }
 
+    /**
+     * Returns the index of x in the set, index start from 0.
+     * If the set doesn't contain x , this function will return -1.
+     * The difference with rank function is that this function will return -1
+     * when x isn't in the set, but the rank function will return a
+     * non-negative number.
+     */
+    int128_t getIndex(uint64_t x) const {
```

Comment:
Sorry, I'm afraid I didn't grasp what you meant...

Reply:
I don't think we need to use int128_t, because if you ever had an item at an index greater than an int64_t can represent, you're already out of memory. The `rank` method returns a uint64_t, and I argue this method should return an int64_t.

The most efficient way to get there is to have a range container for every slot in every bitmap. Each range container can cover 0x10000 items. So to get to 0x8000_0000_0000_0000 (the first index which can't be returned in an i64), you need 0x0000_8000_0000_000 range containers (140737488355328). If each range container (and the keys array, and the container type array, and malloc overhead, and...) magically took just 1 byte per container (spoiler, it will take a LOT more than that), you'd still need 128TiB of memory. There's simply no way to ever realistically have a RoaringBitmap64 with more than 2^63 items.

Reply:
My math is slightly more pessimistic. Each container can represent 65536 values. I get 2**64/(2**16) is  281474976710656. Assuming one byte per container that is... 256 TiB.

I think that Windows cannot ever support more than 6 TB of memory... https://learn.microsoft.com/fr-ca/windows/win32/memory/memory-limits-for-windows-releases?redirectedfrom=MSDN

A reasonable solution here is to use a 64-bit value and explain why you do so.

Reply:
Signed integer means `2^63` rather than `2^64`. I made that mistake initially, you can see I said 256 TB in my first message. Not that it matters, either way, even assuming an impossible 1 byte per container, it's not gonna fit in even virtual memory.

Reply:
Right ok. Most systems have a theoretical limit of maybe 2^56 bytes per process. Of course, this may change in the future, but before that happens, int128_t will be part of the every standard programming language.

Reply:
👍 Thank you all for the professional and detailed explanations. I have changed `int128_t` to `int64_t` in the latest commit.